### DISCUSSION TITLE
access to "app" in "init" functions

### DISCUSSION THREAD

**Discussion by jkdoshi** [pocketbase/pocketbase#712] [Q&A] [answered]
Currently, it is possible to perform migrations in your Go "init" functions by calling m.register.

Can we have a similar thing for app-server init too?

In my code here - https://github.com/spinspire/pocketbase-sveltekit-starter/blob/master/backend/hooks/hooks.go - I have to export a function called `PocketBaseInit` and then call it from `main.go` to register model-record-change hooks (or to register routes).

I do understand why that is not possible in Go package's "init" func (because app is not bootstrapped yet). But wouldn't it be nice if we could somehow setup default listeners for app.OnBeforeServe etc. events beforehand so that _when_  the app-server starts it already have some of these listeners pre-registered? The main benefit being that we won't need to export a function from the extension package, and then call it from main.

**Comment by ganigeorgiev** [comment, marked as answer]
I don't think that's possible without using globals. And even with globals you'll still need to instruct your program to execute the bindings from your package.

Injecting the app instance is the better way to handle this. It is easier to test and allows attaching to multiple PocketBase instances.
